src/useLocalStorage.ts

```diff
@@ -79,7 +79,7 @@ const useLocalStorage = <T>(
         // localStorage can throw. Also JSON.stringify can throw.
       }
     },
-    [key, setState]
+    [key, setState, state]
```

Comment:
I think we need to wrap lines 62 to 76 in a `setState` call, to fix the case of @ribx 
something like this: 

```ts
const set: Dispatch<SetStateAction<T | undefined>> = useCallback(
    (valOrFunc) => {
      try {
        setState(oldValue => {
          const newState =
            typeof valOrFunc === 'function' ? (valOrFunc as Function)(oldState) : valOrFunc;
          if (typeof newState === 'undefined') return;
          let value: string;
          if (options)
            if (options.raw)
              if (typeof newState === 'string') value = newState;
              else value = JSON.stringify(newState);
            else if (options.serializer) value = options.serializer(newState);
            else value = JSON.stringify(newState);
          else value = JSON.stringify(newState);
          localStorage.setItem(key, value);
          return deserializer(value)
        }
      } catch {
        // If user is in private mode or has storage restriction
        // localStorage can throw. Also JSON.stringify can throw.
      }
    },
    [key, setState]
  );
  ```